internal/service/lexv2models/confirmation_setting_equality.go

```diff
@@ -0,0 +1,179 @@
+			!oldConfirmationSettings.DeclinationNextStep.Equal(newConfirmationSettings.DeclinationNextStep) ||
+			!oldConfirmationSettings.DeclinationResponse.Equal(newConfirmationSettings.DeclinationResponse) ||
+			!oldConfirmationSettings.ElicitationCodeHook.Equal(newConfirmationSettings.ElicitationCodeHook) ||
+			!oldConfirmationSettings.FailureConditional.Equal(newConfirmationSettings.FailureConditional) ||
+			!oldConfirmationSettings.FailureNextStep.Equal(newConfirmationSettings.FailureNextStep) ||
+			!oldConfirmationSettings.FailureResponse.Equal(newConfirmationSettings.FailureResponse) {
+			return false, diags
+		}
+
+		oldPromptSpec, di := oldConfirmationSettings.PromptSpecification.ToPtr(ctx)
+		diags = append(diags, di...)
+		if diags.HasError() {
+			return false, diags
+		}
+
+		newPromptSpec, di := newConfirmationSettings.PromptSpecification.ToPtr(ctx)
+		diags = append(diags, di...)
+		if diags.HasError() {
+			return false, diags
+		}
+
+		if !oldPromptSpec.AllowInterrupt.Equal(newPromptSpec.AllowInterrupt) ||
+			!oldPromptSpec.MaxRetries.Equal(newPromptSpec.MaxRetries) ||
+			!oldPromptSpec.MessageGroup.Equal(newPromptSpec.MessageGroup) ||
+			!oldPromptSpec.MessageSelectionStrategy.Equal(newPromptSpec.MessageSelectionStrategy) {
+			return false, diags
+		}
+
+		return arePromptAttemptsEqual(ctx, oldPromptSpec.PromptAttemptsSpecification, newPromptSpec.PromptAttemptsSpecification)
+	}
+
+	return false, diags
+}
+
+// arePromptAttemptsEqual compares two PromptAttemptsSpecification fields for equality
+// treating them as maps with map_block_key as the key
+func arePromptAttemptsEqual(ctx context.Context, oldAttempts, newAttempts fwtypes.SetNestedObjectValueOf[PromptAttemptsSpecification]) (bool, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	// If both are null or unknown, they're equal
+	if oldAttempts.IsNull() && newAttempts.IsNull() {
+		return true, diags
+	}
+	if oldAttempts.IsUnknown() && newAttempts.IsUnknown() {
+		return true, diags
+	}
+
+	if !oldAttempts.Equal(newAttempts) {
+		// Convert to slices for comparison
+		oldPromptAttemptSpecification, di := oldAttempts.ToSlice(ctx)
+		diags = append(diags, di...)
+		if diags.HasError() {
+			return false, diags
+		}
+
+		newPromptAttemptSpecification, di := newAttempts.ToSlice(ctx)
+		diags = append(diags, di...)
+		if diags.HasError() {
+			return false, diags
+		}
+
+		var hasDefaults bool
+		for _, value := range oldPromptAttemptSpecification {
+			key := value.MapBlockKey.ValueString()
+			index := slices.IndexFunc(newPromptAttemptSpecification, func(item *PromptAttemptsSpecification) bool {
+				return item.MapBlockKey.ValueString() == key
+			})
+
+			if index != -1 {
+				if !arePromptAttemptValuesEqual(*newPromptAttemptSpecification[index], *value) {
+					return false, diags
+				}
+			} else if _, ok := promptAttemptSpecificationDefaults(ctx, key); ok {
+				hasDefaults = true
+				continue
+			}
```

Comment:
This will cause the outer function to return `true` if any of the default map block keys are omitted in the "new" configuration (`hasDefaults` isn't updated again even if  `index` == `-1` and `ok` evaluates to `false` on a subsequent value).

Should this change to verify all "missing" blocks match an expected default? Some unit tests may be helpful here given the complexity of the comparison required.

Reply:
@jar-b reworked this to check for all configured arguments and any default attributes that are missing from the configuration.